content/extending-chat-widget/custom-identity-provider/index.mdx

```diff
@@ -0,0 +1,173 @@
+- `getFreshToken` - resolving [Chat Widget token](#chat-widget-token). This should always make a call for fresh token from accounts API.
+- `hasToken` - resolving boolean. Determining whether token has been acquired.
+- `invalidate` - resolving nothing. When called should remove current token. There is no need to do anything else as new token will be requested by `getFreshToken` afterwards.
+
+# Chat Widget Token
+
+LiveChat authorization token you are willing to resolve in functions should contain the following properties:
+
+| Parameter      | Description                                                                      |
+| -------------- | -------------------------------------------------------------------------------- |
+| `accessToken`  | A token you can use to call LiveChat APIs on behalf of the user (customer).      |
+| `entityId`     | The ID of customer.                                                              |
+| `expiresIn`    | A number in **milliseconds** specifying how long the `accessToken` will be valid.|
+| `tokenType`    | Value: `Bearer`.                                                                 |
+| `creationDate` | Unix timestamp specyfing the time of creation of the token.                       |
+| `licenseId`    | LiveChat license ID.                                                             |
+
+## How to acquire it?
+
+In order to get the token you need to implement authorizing function making a call to accounts' `/customer` endpoint. To do so, you can follow [creating new customer flow](https://developers.livechat.com/docs/authorization/authorizing-api-calls/#creating-a-new-customer) or have a look at our [example](https://github.com/livechat/identity-provider-example/blob/master/lib/get-customer-token.ts) using cookies. Mind the fact that the [data](https://developers.livechat.com/docs/authorization/authorizing-api-calls/#response-4) returned from this endpoint is not 1:1 with token shape described above. You will have to add the `creationDate` and `licenseId` fields. Additionally, we expect `expiresIn` to be defined in milliseconds while endpoint operates in seconds, therefore you should just multiply the provided value by `1000` before passing it further.
+
+
+# Examples
+
+Here are our suggested implementations of Customer Identity Provider. The `fetchLiveChatToken()` method used in examples is used to acquire the [Chat Widget token](#chat-widget-token).
+
+## Without caching
+
+If you don't need to cache the token the fetching logic shall occur on every refresh of your application page. In such case the implementation can be very simple - we only have to ensure handling of `getToken` promise and simply resolve others.
+
+```js
+ window.__lc.custom_identity_provider = () => {
+    let tokenPromise = null
+    const fetchToken = () => {
+        tokenPromise = fetchLiveChatToken()
```

Comment:
maybe the `fetchLiveChatToken` could be also explained here, even with the pseudo-code? like, defined above and filled with comments, e.g.:

```js
const fetchLiveChatToken = function() {
        /* fetch a token from LiveChat Accounts API here */
}
```

but a bit more descriptive?